warp10/src/main/java/io/warp10/script/functions/REVERSE.java

```diff
@@ -58,10 +58,8 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
       Collections.reverse((List) top);
       stack.push(top);
     } else if (top instanceof String) {
-      if (!this.stable) {
-        top = new String(UnsafeString.getChars(top.toString()));
-      }
-      char[] chars = UnsafeString.getChars(top.toString());
+      char[] chars = ((String) top).toCharArray();
```

Comment:
won't work with char > 0xFFFF , but anyway. no one will reverse emoji. 
- previous behavior on 2.x was to return  "??" for emoji and it transforms the input. (if you try '💩' DUP REVERSE , you see the input is transformed)
- this behavior output \uD83D\uDCA9  and \uDCA9\uD83D reversed. It is better not to transform the input anymore.

So this is a behavior difference, but acceptable.

Reply:
The behavior is indeed undefined if there are composite UNICODE characters coded on more than two UTF-16 characters. But behavior should be identical to that of 2.x.